vscode/src/completion.ts

```diff
@@ -86,7 +86,15 @@ class QSharpCompletionItemProvider implements vscode.CompletionItemProvider {
       return item;
     });
 
+    let isEmptyOrWhitespace = true;
+    for (let i = 0; i < document.lineCount; i++) {
+      if (!document.lineAt(i).isEmptyOrWhitespace) {
+        isEmptyOrWhitespace = false;
+        break;
+      }
+    }
+
     // Include the samples list for empty documents
-    return document.lineCount > 0 ? results : results.concat(this.samples);
+    return !isEmptyOrWhitespace ? results : results.concat(this.samples);
```

Comment:
I'm not sure this is the right approach. What if I start a file with some TODO comments (which I often do) or a license header (which is common)? Or I import some items knowing I'll be using them as I expand on the sample I'm starting from?

I think it's reasonable to only show when the user is on a blank line though.

Reply:
Actually, in the spirit of the 'correctness', being that the samples are all 'top level' code, could we just show them in situations when at the top level (i.e. the places where we currently show 'namespace').

I'd rather limit the 'magic' of when samples do or don't appear in a completion list that may not be intuitive to users. Have them appear only at top level seems to fit in the with correctness goal of only having  completions show where they are valid.

Reply:
> i.e. the places where we currently show 'namespace' 

Oh, now I think I see what you were getting at. I'll just check for the presence of `namespace` in the list, should be trivial and fairly accurate :)

Reply:
Per offline discussions. Will change "namespace" to "operation" in anticipation of #2023